redfish-core/lib/led.hpp

```diff
@@ -282,23 +314,59 @@ inline void
 {
     BMCWEB_LOG_DEBUG << "Set LocationIndicatorActive";
 
-    dbus::utility::getAssociationEndPoints(
-        objPath + "/identifying",
-        [aResp, ledState](const boost::system::error_code& ec,
-                          const dbus::utility::MapperEndPoints& endpoints) {
+    // get the association fType
+    sdbusplus::asio::getProperty<
+        std::vector<std::tuple<std::string, std::string, std::string>>>(
+        *crow::connections::systemBus, "xyz.openbmc_project.Inventory.Manager",
+        objPath, "xyz.openbmc_project.Association.Definitions", "Associations",
+        [aResp, ledState,
+         objPath](const boost::system::error_code ec,
+                  std::vector<std::tuple<std::string, std::string, std::string>>
+                      associations) {
         if (ec)
         {
-            if (ec.value() != EBADR)
-            {
-                messages::internalError(aResp->res);
-            }
+            BMCWEB_LOG_ERROR << "DBUS response error " << ec;
+            return;
+        }
+        if (associations.empty())
+        {
+            BMCWEB_LOG_ERROR << "Associations not found for - [ " << objPath
+                             << " ]. Skipping Set LED.";
             return;
         }
+        // TODO: check we want identifying OR fault_identifying??
+        // or it varies with object?
+        // because identifying is not available for object system.
+        std::string fType{get<0>(associations[0])};
 
-        for (const auto& endpoint : endpoints)
+        // validate the fType
+        if (fType.empty())
         {
-            setLedAsset(aResp, endpoint, ledState);
+            BMCWEB_LOG_ERROR << "Association Forward Type is not found for - [ "
+                             << objPath << " ]. Skipping Set LED.";
+            return;
         }
+
+        // get Association endPoints
+        dbus::utility::getAssociationEndPoints(
+            objPath + "/" + fType,
```

Comment:
Even if there are more than 1 associations, the code is treating all of those associations for LED.
I don't think this seems right.